text/0763-asset-importing-spec.md

```diff
@@ -86,20 +94,51 @@ HTML | Other | `<img src="./thing.png" />` | W3C
     }
     ```
 
-3. Importing any "Other Asset" file gives you back a `default` export containing a valid URL to that file.
+3. Calling `import.meta.resolve` with any "Other Asset" file gives you back a string containing a valid URL for that file.
+
+### `import.meta.resolve(moduleName)`
+
+`import.meta.resolve(moduleName)`[^1] is a function built-in to browsers that takes a `moduleName` and "returns a string corresponding to the path that would be imported if the argument were passed to import()." [^2]
+
+The build system will handle uses of `import.meta.resolve` when called with assets other than `.js` or .css` (see above).
+
+The argument `moduleName` supports a subset of dynamicism as defined [here](https://github.com/emberjs/rfcs/blob/master/text/0507-embroider-v2-package-format.md#supported-subset-of-dynamic-import-syntax).
+
+The argument `moduleName` must explicitly include the extension.
+
+For example, the following is valid as the build system can determine the files that possibly match the interpolation:
+
+```js
+class extends Component {
+  get iconURL() {
+    return import.meta.resolve(`../icons/${this.args.locale}.png`)
+  }
+}
+```
+
+Calling `import.meta.resolve` with an asset that does not exist will cause a build error. You cannot fail to notice if you accidentally delete an asset that is still in use somewhere in your app.
+
+#### Isn't it risky to "take over" a browser API?
+
+Yes and no. This [original proposal for this feature to WHATWG](https://github.com/whatwg/html/issues/3871#issue-346547968) envisioned build tools rewriting the expressions to provide asset optimizations.
```

Comment:
What's the risk that "Yes" is referring to? I read the following text as arguing against the existence of risk, or do I misunderstand this?

Reply:
No, just poor wording.